vertical-module/templates/client/controllers/_.client.controller.js

```diff
@@ -25,18 +25,33 @@ angular.module('<%= slugifiedPluralName %>').controller('<%= classifiedPluralNam
 
 		// Remove existing <%= humanizedSingularName %>
 		$scope.remove = function( <%= camelizedSingularName %> ) {
-			if ( <%= camelizedSingularName %> ) { <%= camelizedSingularName %>.$remove();
+			$modal.open({
+				templateUrl: 'modules/<%= slugifiedPluralName %>/views/remove-<%= slugifiedSingularName %>.client.view.html',
+				controller: function($scope, $modalInstance) {
+					$scope.ok = function() {
+						$modalInstance.close();
+					};
 
-				for (var i in $scope.<%= camelizedPluralName %> ) {
-					if ($scope.<%= camelizedPluralName %> [i] === <%= camelizedSingularName %> ) {
-						$scope.<%= camelizedPluralName %>.splice(i, 1);
+					$scope.cancel = function() {
+						$modalInstance.dismiss('cancel');
+					};
+				}
+			})
+			.result.then(function() {
+				if ( <%= camelizedSingularName %> ) {
+					<%= camelizedSingularName %>.$remove();
+
+					for (var i in $scope.<%= camelizedPluralName %> ) {
+						if ($scope.<%= camelizedPluralName %> [i] === <%= camelizedSingularName %> ) {
```

Comment:
Where? I just wrapped the already existing code within the .result.then(function(){ }); block, and all the spaces are actually tabs. If this is not the standard way of doing it, I'm ok following any other guidelines if there are. I'm new in the nodejs world (less than 2 months, though over 7 years in other technologies), so I'm not ocmpletely familiar with the standards yet.